src/prpy/planning/vectorfield.py

```diff
@@ -197,32 +217,165 @@ def TerminateMove():
+
+        def TerminateMove():
+            """
+            Function defining the termination condition.
+
+            Fail if deviation larger than position and angular tolerance.
+            Succeed if distance moved is larger than max_distance.
+            Cache and continue if distance moved is larger than distance.
+            """
+            from .exceptions import ConstraintViolationPlanningError
+
+            T_ee_curr = manip.GetEndEffectorTransform()
+
+            # Find where we are on the goal trajectory by finding
+            # the the closest point
+            (_, t) = util.GetMinDistanceBetweenTransformAndWorkspaceTraj(
+                                                                     T_ee_curr,
+                                                                     traj)
+            # Get the desired end-effector transform from
+            # the goal trajectory
+            T_ee_desired = openravepy.matrixFromPose(traj.Sample(t)[0:7])
+
+            # Calculate error between current end-effector pose
+            # and where we should be on the goal trajectory
+            geodesic_error = util.GeodesicError(T_ee_desired, T_ee_curr)
+            orientation_error = geodesic_error[3]
+            position_error = geodesic_error[0:3]
+
+            if numpy.fabs(orientation_error) > angular_tolerance:
+                raise ConstraintViolationPlanningError(
+                    'Deviated from orientation constraint.')
+
+            position_deviation = numpy.linalg.norm(position_error)
+            if position_deviation > position_tolerance:
+                raise ConstraintViolationPlanningError(
+                    'Deviated from straight line constraint.')
+
+            # Check if we have reached the end of the goal trajectory
+            dist_to_goal = util.GetGeodesicDistanceBetweenTransforms(T_ee_curr,
+                                                                     T_ee_goal)
+            if numpy.fabs(dist_to_goal) <= goal_pose_tol:
+                return Status.CACHE_AND_TERMINATE
+
+            return Status.CONTINUE
+
+        integration_timelimit = 10.0
+        return self.FollowVectorField(robot, vf_path, TerminateMove,
+                                      integration_timelimit,
                                       timelimit, **kw_args)
 
+
     @PlanningMethod
     def FollowVectorField(self, robot, fn_vectorfield, fn_terminate,
                           integration_timelimit=10.,
-                          timelimit=5.0, dt_multiplier=1.01, **kw_args):
+                          timelimit=5.0,
+                          **kw_args):
         """
         Follow a joint space vectorfield to termination.
 
         @param robot
         @param fn_vectorfield a vectorfield of joint velocities
         @param fn_terminate custom termination condition
+        @param integration_timelimit The initial time step to be taken
+                                     by the integrator.
```

Comment:
This `__docstring__` is misleading. The initial step size is passed as the `first_step` parameter to the integrator and is currently hard-coded to 0.1. This argument appears to be the value of `t` for which we're trying to compute the value of the integral.